packages/front-end/services/datasources.ts

```diff
@@ -257,6 +257,53 @@ FROM
   },
 };
 
+const RudderstackSchema: SchemaInterface = {
+  experimentDimensions: ["device", "browser"],
+  getExperimentSQL: (tablePrefix, userId) => {
+    return `SELECT
+  ${userId},
+  received_at as timestamp,
+  experiment_id,
+  variation_id,
+  (CASE
+    WHEN context_user_agent LIKE '%Mobile%' THEN 'Mobile'
+    ELSE 'Tablet/Desktop' END
+  ) as device,
+  (CASE 
+    WHEN context_user_agent LIKE '% Firefox%' THEN 'Firefox'
+    WHEN context_user_agent LIKE '% OPR%' THEN 'Opera'
+    WHEN context_user_agent LIKE '% Edg%' THEN ' Edge' 
+    WHEN context_user_agent LIKE '% Chrome%' THEN 'Chrome'
+    WHEN context_user_agent LIKE '% Safari%' THEN 'Safari'
+    ELSE 'Other' END
+  ) as browser
+FROM
+  ${tablePrefix}experiment_viewed
+WHERE
+  ${userId} is not null`;
+  },
+  getIdentitySQL: (tablePrefix) => {
+    return [
```

Comment:
Return an empty array here.  This is for joining between different identifier types and doesn't make sense when there's only 1.